src/js/models/layer.js

```diff
@@ -680,7 +683,9 @@ define(function (require, exports, module) {
                 vectorMaskEmpty: layerDescriptor.vectorMaskEmpty,
                 textWarningLevel: layerDescriptor.textWarningLevel,
                 isLinked: _extractIsLinked(layerDescriptor),
-                initialized: initialized || selected
+                boundsInitialized: layerDescriptor.boundsInitialized,
+                initialized: layerDescriptor.initialized,
+                descriptor: Immutable.Map(layerDescriptor)
```

Comment:
Keep the original descriptor so that we can reset the layer from a partial layer descriptor. For example, only resetting the layer bounds.